src/ccsr/src/client.rs

```diff
@@ -364,6 +397,62 @@ impl fmt::Display for GetByIdError {
     }
 }
 
+#[derive(Debug, Deserialize)]
+#[serde(rename_all = "camelCase")]
+pub struct SubjectConfig {
+    pub compatibility_level: CompatibilityLevel,
+    // There are other fields to include if we need them.
+}
+
+/// Errors for schema lookups by subject.
+#[derive(Debug)]
+pub enum GetSubjectConfigError {
+    /// The requested subject does not exist.
+    SubjectNotFound,
+    /// The underlying HTTP transport failed.
+    Transport(reqwest::Error),
+    /// An internal server error occurred.
+    Server { code: i32, message: String },
+}
+
+impl From<UnhandledError> for GetSubjectConfigError {
+    fn from(err: UnhandledError) -> GetSubjectConfigError {
+        match err {
+            UnhandledError::Transport(err) => {
+                if err.status() == Some(reqwest::StatusCode::NOT_FOUND) {
+                    GetSubjectConfigError::SubjectNotFound
+                } else {
+                    GetSubjectConfigError::Transport(err)
+                }
```

Comment:
Are you sure about this? I don't think 4xx errors come through as transport errors. I would expect this to look more like the other `From<UnhandledError>` implementations where you need to match on `UnhandledError::Api.code`. 

Also note that the error I'm seeing when a subject does not exist has code 40408 and message "Subject does not have subject-level compatibility configured". I think the CSR doesn't distinguish between "subject does not exist" and "subject does not have compatibility configured".

Reply:
You are totally right - this returns a 40408 when the subject level is not yet set. Adding the tests caught this. Undocumented behavior!

Reply:
Ah, it's not undocumented, it's that `reqwest` by default won't throw an error on a 4xx or 5xx error code. You need to call `raise_for_status()` if you want that behavior. In this case, the code is structured so that we _don't_ want that behavior, because `send_request` is already doing all the right things and pulling out the more specific error code (40408) from the response body, vs the less specific 404 status code of the HTTP response itself.

Reply:
It's undocumented in the linked confluent API docs! All the other APIs describe the internal codes used for specific error types, but the Get Config docs just show this:
![image](https://github.com/MaterializeInc/materialize/assets/3818834/29cafcf5-fa4d-4d67-a6b0-ff03dfb61dec)
despite it actually returning the 40408 error when the subject isnt set

Reply:
Oh sorry I see, I misunderstood! Yes, huh, that does seem to be undocumented.

It looks like you've still got the 404 check in the branch above—I think we should just remove that since we'll never create a transport error with status 404!